### PR TITLE
Check Khoury webhook for signature

### PR DESCRIPTION
**Pull request by willstenzel**

### CODE REVIEW

File: packages/server/src/login/khoury-public-key.pem [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+-----BEGIN PUBLIC KEY-----
+MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDCFENGw33yGihy92pDjZQhl0C3
+6rPJj+CvfSC8+q28hxA161QFNUd13wuCTUcq0Qd2qsBe/2hFyc2DCJJg0h1L78+6
+Z4UMR7EOcpfdUE9Hf3m/hs+FUR45uBJeDK1HSFHD8bHKD6kv8FPGfJTotc+2xjJw
+oYi+1hqp1fIekaxsyQIDAQAB
+-----END PUBLIC KEY-----
```
Comment (willstenzel) [inline packages/server/src/login/khoury-public-key.pem:7]: This is a place-holder public key. We'll have to replace with the real one when Khoury hooks it up on their end

File: packages/server/src/login/http-signature.service.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@nestjs/common';
+import * as fs from "fs";
+import * as httpSignature from 'http-signature';
+import { Request } from 'express';
+
+
+@Injectable()
+export class HttpSignatureService {
+    public verifyRequest(req: Request): boolean {
+        const PUBLIC_KEY = fs.readFileSync(__dirname + '/khoury-public-key.pem', 'ascii');
```
Comment (willstenzel) [inline packages/server/src/login/http-signature.service.ts:10]: Not sure if this could be made a class variable so for better performance?

**Comment by willstenzel** [comment]
I'm planning to add a test in the `http-signature.service.spec.ts` before marking it "Ready for Review."

File: packages/server/src/login/http-signature.service.ts (hunk 2 above)
Comment (dajinchu) [reply packages/server/src/login/http-signature.service.ts:10]: Yes please do this in the constructor

Comment (willstenzel) [reply packages/server/src/login/http-signature.service.ts]: done

**Comment by willstenzel** [comment]
> I'm planning to add a test in the `http-signature.service.spec.ts` before marking it "Ready for Review."

Actually I'm not so sure the best way to write a test for this. 

In the example [here](https://github.com/joyent/node-http-signature#client) they use another library called `https` but not sure we want to install another dependency just for a test. Any help would be appreciated.

File: packages/server/src/login/http-signature.service.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import { Request } from 'express';
+
+
+@Injectable()
+export class HttpSignatureService {
+    PUBLIC_KEY: string;
+    constructor() {
+        this.PUBLIC_KEY = fs.readFileSync(__dirname + '/khoury-public-key.pem', 'ascii');
+    }
+    public verifyRequest(req: Request): boolean {
```
Comment (dajinchu) [inline packages/server/src/login/http-signature.service.ts:13]: I think we should just put this in the controller. Services should deal with the business logic, but I think this is more of an HTTP-layer concern.

Comment (dajinchu) [reply packages/server/src/login/http-signature.service.ts:13]: Actually, leaving it in the service might make testing more feasible. I think using the `https` or `http` lib is fine - they're actually part of the node standard library so we don't need to add any packages.

File: packages/server/src/login/login.controller.ts [hunk 4]
```diff
@@ -19,15 +21,23 @@ import {
...
 import { ConfigService } from '@nestjs/config';
 import { LoginCourseService } from './login-course.service';
 import { CourseSectionMappingModel } from './course-section-mapping.entity';
+import * as httpSignature from 'http-signature';
+import * as path from 'path';
+import * as fs from 'fs';
 
 @Controller()
 export class LoginController {
+  PUBLIC_KEY = fs.readFileSync(
```
Comment (liustanley) [inline packages/server/src/login/login.controller.ts:36]: i think cypress is erroring on this line just fyi

**Review by liustanley — APPROVED** [review]
lgtm!

File: packages/server/src/login/login.controller.ts (hunk 4 above)
Comment (willstenzel) [reply packages/server/src/login/login.controller.ts]: thanks for the heads up!

**Pull request merged**